src/Fixer/Phpdoc/PhpdocListTypeFixer.php

```diff
@@ -0,0 +1,129 @@
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        $codeSample = <<<'PHP'
+            <?php
+            /**
+             * @param bool[] $a
+             * @param array<int> $b
+             * @param array<string, int> $c
+             * @param list<int> $d
+             */
+
+            PHP;
+
+        return new FixerDefinition(
+            'PHPDoc list types must be written in configured style.',
+            [
+                new CodeSample($codeSample),
+                new CodeSample($codeSample, ['style' => self::STYLE_ARRAY]),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before PhpdocAlignFixer, PhpdocTypesOrderFixer.
+     * Must run after AlignMultilineCommentFixer, CommentToPhpdocFixer, PhpdocIndentFixer, PhpdocScalarFixer, PhpdocToCommentFixer, PhpdocTypesFixer.
+     */
+    public function getPriority(): int
+    {
+        return 1;
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('style', sprintf('Whether to use `%s` or `%s` as type.', self::STYLE_ARRAY, self::STYLE_LIST)))
+                ->setAllowedValues([self::STYLE_ARRAY, self::STYLE_LIST])
+                ->setDefault(self::STYLE_LIST)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind([T_DOC_COMMENT])) {
+                continue;
+            }
+
+            $docBlock = new DocBlock($tokens[$index]->getContent());
+
+            foreach ($docBlock->getAnnotations() as $annotation) {
+                if (!$annotation->supportTypes()) {
+                    continue;
+                }
+
+                $typeExpression = $annotation->getTypeExpression();
+                if (null === $typeExpression) {
+                    continue;
+                }
+
+                $type = $typeExpression->toString();
+                $type = $this->fixType($type);
+                if (self::STYLE_LIST === $this->configuration['style']) {
+                    $type = Preg::replace('/array(?=<[^,]+(>|<|{|\\())/', self::STYLE_LIST, $type);
+                }
+                $annotation->setTypes([$type]);
+            }
+
+            $newContent = $docBlock->getContent();
+            if ($newContent === $tokens[$index]->getContent()) {
+                continue;
+            }
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $newContent]);
+        }
+    }
+
+    private function fixType(string $type): string
+    {
+        $newType = Preg::replace('/([\\\\a-zA-Z0-9_>]+)\[\]/', 'array<$1>', $type);
+
+        if ($newType !== $type) {
+            return $this->fixType($newType);
```

Comment:
Maybe it's a bit of an extreme situation but wouldn't it be better to use a loop to avoid too much recursion?

Reply:
Very valid point, it was my laziness to use the recursion here.